neo/rawio/baserawio.py

```diff
@@ -587,8 +602,13 @@ def get_analogsignal_chunk(self, block_index=0, seg_index=0, i_start=None, i_sto
 
         return raw_chunk
 
-    def rescale_signal_raw_to_float(self, raw_signal, dtype='float32', stream_index=None,
-                                    channel_indexes=None, channel_names=None, channel_ids=None):
+    def rescale_signal_raw_to_float(self,
+                                    raw_signal: np.ndarray,
+                                    dtype: np.dtype = 'float32',
```

Comment:
shouldn't this be `Union[np.dtype, str]`, since `'float32'` is a string?

Reply:
I thought np.dtype included "float32" since numpy accepts float32? But maybe I'm wrong. We don't accept any string (for example I can do "float100000") so I think the only floats accepted are numpy floats which are contained by the numpy.dtype.

Reply:
I don't have a very deep understanding of type hints in Python, but my reasoning was based on this:
```
In [1]: import numpy as np

In [2]: x = np.arange(5)

In [3]: x.dtype
Out[3]: dtype('int64')

In [4]: isinstance(x.dtype, np.dtype)
Out[4]: True

In [5]: isinstance("int64", np.dtype)
Out[5]: False
```

Reply:
That's a great point I meant more in the sense that you can add `int64` as the dtype in any np.function.

```
import numpy as np

>>> x = np.arange(5, dtype='int64')
>>> x.dtype
dtype('int64')

>>> x = np.arange(5, dtype='int2')
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
TypeError: data type 'int2' not understood
```

It won't accept any string, but only the strings allowable by the numpy.dtype constructor. So you're right in that it is a string as its type, but it's only the strings accepted by numpy. Maybe there is a better way to represent that, but I thought the most succinct (and generally understandable) way to do this would be to tell people to just use `np.type`. I can do some reading and see if there is a more standard way to represent this though!

Reply:
Looking into numpy's own typing, it looks like maybe we should use `import numpy.typing as npt` and 
```dtype: npt.DTypeLike = 'float32'```

Reply:
I just put the same comment below :) haha.

Reply:
sorry, just saw you posted the same thing below.